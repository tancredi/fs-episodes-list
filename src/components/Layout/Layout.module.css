.background {
  position: fixed;
  width: 110%;
  height: 110%;
  opacity: 0.1;
  filter: blur(5px);
  object-fit: cover;
  margin: -5%;
}

.page {
  position: relative;
  z-index: 1;
  max-width: 900px;
  margin: auto;
  padding: 0 30px;
  box-sizing: border-box;
}

.logo {
  width: 300px;
  display: block;
  margin: 60px auto;
}
