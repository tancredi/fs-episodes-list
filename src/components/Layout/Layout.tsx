import type { FunctionComponent, PropsWithChildren } from "react";

export const Layout: FunctionComponent<PropsWithChildren<{}>> = ({
  children,
}) => <>{children}</>;
