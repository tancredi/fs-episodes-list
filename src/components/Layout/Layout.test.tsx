import React from "react";
import { render } from "@testing-library/react";
import { Layout } from "./Layout";

describe("Layout component", () => {
  it("renders UI correctly", () => {
    expect(render(<Layout>Foo bar</Layout>).asFragment()).toMatchSnapshot();
  });

  it("renders `children` prop", () => {
    const children = "foo";

    render(<Layout>{children}</Layout>).getByText(children);
  });

  it("renders a logo with correct alt text", () => {
    expect(render(<Layout />).getByAltText("Final Space logo").tagName).toBe(
      "IMG"
    );
  });
});
