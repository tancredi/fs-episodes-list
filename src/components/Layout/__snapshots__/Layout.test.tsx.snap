// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders UI correctly 1`] = `
<DocumentFragment>
  <img
    class="background"
    src="/mock/asset/space-background@2x.jpg"
  />
  <div
    class="page"
  >
    <img
      alt="Final Space logo"
      class="logo"
      src="/mock/asset/logo@2x.png"
    />
    Foo bar
  </div>
</DocumentFragment>
`;
